userspace/falco/stats_writer.cpp

```diff
@@ -161,39 +182,219 @@ void stats_writer::worker() noexcept
 	}
 }
 
-stats_writer::collector::collector(std::shared_ptr<stats_writer> writer)
-	: m_writer(writer), m_last_tick(0), m_samples(0)
+stats_writer::collector::collector(const std::shared_ptr<stats_writer>& writer)
+	: m_writer(writer), m_last_tick(0), m_samples(0),
+	  m_last_now(0), m_last_n_evts(0), m_last_n_drops(0), m_last_num_evts(0)
 {
+}
+
+void stats_writer::collector::get_metrics_output_fields_wrapper(
+		std::unordered_map<std::string, std::string>& output_fields,
+		const std::shared_ptr<sinsp>& inspector, uint64_t now,
+		const std::string& src, uint64_t num_evts, double stats_snapshot_time_delta_sec)
+{
+	static const char* all_driver_engines[] = {
+		BPF_ENGINE, KMOD_ENGINE, MODERN_BPF_ENGINE,
+		SOURCE_PLUGIN_ENGINE, NODRIVER_ENGINE, UDIG_ENGINE, GVISOR_ENGINE };
+	const scap_agent_info* agent_info = inspector->get_agent_info();
+	const scap_machine_info* machine_info = inspector->get_machine_info();
+
+	/* Wrapper fields useful for statistical analyses and attributions. Always enabled. */
+	output_fields["evt.time"] = std::to_string(now); /* Some ETLs may prefer a consistent timestamp within output_fields. */
+	output_fields["version"] = FALCO_VERSION;
+	output_fields["start_ts"] = std::to_string(agent_info->start_ts_epoch);
+	output_fields["duration_sec"] = std::to_string((now - agent_info->start_ts_epoch) / ONE_SECOND_IN_NS);
+	output_fields["kernel_release"] = agent_info->uname_r;
+	output_fields["host_boot_ts"] = std::to_string(machine_info->boot_ts_epoch);
+	output_fields["hostname"] = machine_info->hostname; /* Explicitly add hostname to log msg in case hostname rule output field is disabled. */
+	output_fields["host_num_cpus"] = std::to_string(machine_info->num_cpus);
 
+	output_fields["src"] = src;
```

Comment:
Starting from this release we'll have a new field named `evt.source` coming from libsinsp, so just like you did with `evt.time`, I'd try to be consistent.
```suggestion
	output_fields["evt.source"] = src;
```